src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -69,27 +84,43 @@
     }
 
     public static @Nonnull GitHub connect(@CheckForNull String apiUri, @CheckForNull StandardCredentials credentials) throws IOException {
```

Comment:
so maybe just `return GitHubPlugin.configuration().findGithubConfig(and(withHost(apiUrl), server -> StringUtils.equals(server.getCredentialsId(), credentials.getId()))` ?
(with replacement of lambda to guava Predicate)